src/nextjournal/clerk/analyzer.clj

```diff
@@ -85,7 +85,9 @@
 #_(rewrite-defcached '(nextjournal.clerk/defcached foo :bar))
 
 (defn deflike? [form]
-  (and (seq? form) (symbol? (first form)) (str/starts-with? (name (first form)) "def")))
+  (and (seq? form)
+       (symbol? (first form))
+       (contains? #{'def 'defonce 'defn} (first form))))
```

Comment:
The intent here was to actually account for user-defined macros that are like `def`s, so more than just those three concrete instances.

Reply:
we could check the returned value is a var of the same name as the one inferred by the analyzer, to decide when to do the wrapping.